v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -0,0 +1,179 @@
+
+- Length of the `file` field name + 1.
+
+- Length of the `func` field name + 1.
+
+- Length of the `line` field name + 1.
+
+- For each property in `log_context`, the length of the property name + 1 and one extra byte for the type of the field.
+
+- For struct properties one extra byte shall be added for the field count.
+
+`log_sink_etw.log_sink_log` shall fill a `SELF_DESCRIBED_EVENT` structure, setting the following fields:
+
+- `_tlgBlobTyp` shall be set to `_TlgBlobEvent4`.
+
+- `_tlgChannel` shall be set to 11.
+
+- `_tlgLevel` shall be set to the appropriate logging level.
+
+- `_tlgOpcode` shall be set to 0.
+
+- `_tlgKeyword` shall be set to 0.
+
+- `_tlgEvtMetaSize` shall be set to the computed metadata size + 4.
+
+- `_tlgEvtTag` shall be set to 128.
+
+`log_sink_etw.log_sink_log` shall fill the event metadata:
+
+- The string `content` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+
+- The string `file` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+
+- The string `func` (as field name, excluding zero terminator), followed by one byte with the value `TlgInANSISTRING`.
+
+- The string `line` (as field name, excluding zero terminator), followed by one byte with the value `TlgInINT32`.
+
+For each property in `log_context` the following shall be added to the event metadata:
+
+- A string with the property name (excluding zero terminator)
+
+- A byte with the type of property, as follows:
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_ascii_char_ptr`, a byte with the value `TlgInANSISTRING` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int64_t`, a byte with the value `TlgInINT64` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint64_t`, a byte with the value `TlgInUINT64` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int32_t`, a byte with the value `TlgInINT32` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint32_t`, a byte with the value `TlgInUINT32` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int16_t`, a byte with the value `TlgInINT16` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint16_t`, a byte with the value `TlgInUINT16` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_int8_t`, a byte with the value `TlgInINT8` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_uint8_t`, a byte with the value `TlgInUINT8` shall be added in the metadata.
+
+  - If the property type is `LOG_CONTEXT_PROPERTY_TYPE_struct`, a byte with the value `_TlgInSTRUCT | _TlgInChain` shall be added in the metadata.
+
+  - If the property type is any other value, it shall be ignored.
```

Comment:
>it shall be ignored. [](http://example.com/codeflow?start=45&length=20)

wouldn't that introduce some unparseable metadata?

Maybe just bail out at this moment?

Reply:
Changed to:
"If the property type is any other value, no property data shall be added to the event."
Let's at least log what we can. We could even improve to log some extra field saying we crapped out.